src/videopython/utils/image.py

```diff
@@ -0,0 +1,171 @@
+        if max_width is None and max_height is None:
+            raise ValueError("You need to pass max_width or max_height")
+        if max_width is not None:
+            width_font_size = self._fit_font_width(text, font, max_width)
+        if max_height is not None:
+            height_font_size = self._fit_font_height(text, font, max_height)
+        return min([size for size in [width_font_size, height_font_size] if size is not None])
+
+    def write_text(
+        self,
+        text: str,
+        font_filename: str,
+        xy: tuple[int, int],
+        font_size: int | None = 11,
+        color: tuple[int, int, int] = (0, 0, 0),
+        max_width: int | None = None,
+        max_height: int | None = None,
+    ) -> tuple[int, int]:
+        x, y = xy
+        if font_size is None and (max_width is None or max_height is None):
+            raise ValueError(f"Must set either `font_size`, or both `max_width` and `max_height`!")
+        elif font_size is None:
+            font_size = self._get_font_size(text, font_filename, max_width, max_height)
+        text_size = self.get_text_size(font_filename, font_size, text)
+        if (text_size[0] + x > self.image_size[0]) or (text_size[1] + y > self.image_size[1]):
+            raise OutOfBoundsError(f"Font size `{font_size}` is too big, text won't fit!")
+        font = ImageFont.truetype(font_filename, font_size)
+        self._draw.text((x, y), text, font=font, fill=color)
+        return text_size
+
+    def get_text_size(self, font_filename: str, font_size: int, text: str) -> tuple[int, int]:
+        """Return bounding box size of the rendered `text` with `font_filename` and `font_size`."""
+        font = ImageFont.truetype(font_filename, font_size)
+        return font.getbbox(text)[2:]
+
+    def _split_lines_by_width(
+        self,
+        text: str,
+        font_filename: str,
+        font_size: int,
+        box_width: int,
+    ) -> list[str]:
+        """Split the `text` into lines of maximum `box_width`."""
+        words = text.split()
+        split_lines: list[list[str]] = []
+        current_line: list[str] = []
+        for word in words:
+            new_line = " ".join(current_line + [word])
+            size = self.get_text_size(font_filename, font_size, new_line)
+            if size[0] <= box_width:
+                current_line.append(word)
+            else:
+                split_lines.append(current_line)
+                current_line = [word]
+        if current_line:
+            split_lines.append(current_line)
+        lines = [" ".join(line) for line in split_lines]
+        return lines
+
+    def write_text_box(
+        self,
+        text: str,
+        font_filename: str,
+        xy: tuple[int, int],
+        box_width: int,
+        font_size: int = 11,
+        color: tuple[int, int, int] = (0, 0, 0),
+        place: Literal["left", "right", "center"] = "left",
+    ) -> tuple[int, int]:
+        x, y = xy
```

Comment:
Could you add some comment how to interpret `x` and `y` for different `place` options?